docs/developer/pull-requests.md

```diff
@@ -0,0 +1,128 @@
+
+That said, beware that not all PRs are created equal. It doesn’t produce the same cognitive load to review a PR with 10 new lines of actual code and 500 lines of tests and data files than 200 lines of actual code changes over 20 different files. Prioritize making the review easier, but don’t create extra work for yourself just to align with a random self imposed number. When in doubt, talk to your reviewers and find something that you both feel comfortable with.
+
+### PR descriptions
+
+This is the best way to help the reviewer build context about the changes in an asynchronous way. Not only that but it serves in a way as documentation for both engineers when tracking down a commit in the future and for outside contributors to get a feel for what a change is/does, how to get involved with it, etc.
+
+A good PR description should include:
+1. A summary of intent including the problem you're solving. The summary may not be fulfilled entirely by this PR but by a series of PRs and the summary should indicate that.
+2. Dependent on the PR, a slightly more detailed description about what you're specifically addressing in said PR.
+3. What's next? Tell the reviewers what your future intentions are (if it applies).
+
+Here are some recommendations to follow while writing the description:
+
+* Follow the Single Responsibility Principle. If you writing a PR description and you realize that you are describing multiple "responsibilities", that's a good sign you should probably split it.
+* Assume the reviewer knows nothing about your work.
+* Try to look at your PR from a reviewer perspective: what would you need to be able to review that PR successfully? Common topics:
+    * What problem are we solving
+    * Why are we solving it
+    * How are we doing it
+* Be explicit, even if you think certain information can be inferred from the code, don’t rely on that. Specify everything that is not obvious.
+* If they exists, link related GitHub issue/s.
+* If they exist (and they are public), link design docs.
+* Reflect all the information that is not documented anywhere else:
+    * Decisions made in other PRs/issues/private conversations
+    * New requirements
+    * How does this PR fit the whole project/feature? Very important if splitting something bigger into small PRs. It helps the reviewer fill the gaps and look at you high level solution without having all the code in front of them.
+
+Same as in the previous point, not all PRs are the same. There are some times when code is self-explanatory, specially when paired with a good commit message. Be careful with this but also exploit it and avoid extra work when you are confident you can.
+
+`TODO: add concrete example for a good PR description`
+
+### Review your own changes
+
+This can dramatically speed up reviews as you're putting yourself in your reviewers shoes. Use the PR comments if you want to add extra context around specific pieces of code.
+
+`TODO: complete this section`
```

Comment:
@danbudris @chrisdoherty4 looking at you both again since you proposed this section